### DISCUSSION TITLE
Atypical(?) Gentoo Linux User Installation Issues first time in 9 years

### DISCUSSION THREAD

**Discussion by BlueStilton** [senshu/Sozi#654] [Q&A]
Dear sozi community,

I'm Frank Mariën (lp@tigerbear.io), a long time Gentoo Linux user, and Sozi enthousiast since about 2013.

Disclosure: I'm an old git that hates all things cloud, containers, etc.. Thinks software should be distributed only as source code, javascript belong on the client-side only, etc.. etc.. I have made an exception for Sozi because I love it that much! This means I don't have/want to infect my system with things like like Docker or AppImage and I reluctantly install npm ONLY because.. Sozi..

I've always managed to get Sozi working somehow, probably because there were general Linux builds that
I could just unpack in /opt/sozi and make a symlink to, with minimal hassle.

But today, after 9 years.. I'm stumped.. 

https://github.com/sozi-projects/Sozi/releases/tag/v22.03.25

has no general Linux archive, so I'm left with building from source.. I get as far as 

`nmp install`

but then I'm faced with all sorts of what I think is npm-specific crud that I have no intention to learn about because (see above ; I do not condone even the existence of npm!)

I need to use 'gulp' whatever that is.

```
frank@minerva:~/src/ext/Sozi (master) $ npm install

npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: '@es-joy/jsdoccomment@0.10.8',
npm WARN EBADENGINE   required: { node: '^12 || ^14 || ^16' },
npm WARN EBADENGINE   current: { node: 'v18.6.0', npm: '8.13.2' }
npm WARN EBADENGINE }
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: 'eslint-plugin-jsdoc@36.1.1',
npm WARN EBADENGINE   required: { node: '^12 || ^14 || ^16' },
npm WARN EBADENGINE   current: { node: 'v18.6.0', npm: '8.13.2' }
npm WARN EBADENGINE }
up to date, audited 1184 packages in 9s
86 packages are looking for funding
  run `npm fund` for details
22 vulnerabilities (10 moderate, 12 high)
To address issues that do not require attention, run:
  npm audit fix
To address all issues possible (including breaking changes), run:
  npm audit fix --force
Some issues need review, and may require choosing
a different dependency.
Run `npm audit` for details.
```

If I look at the AUR package, which builds from source, it's thousands of lines with very specific dependencies, I'm not really encouraged to try using that as instructions.. or even contribute and maintain a Gentoo ebuild!

Need Help! Willing to try stuff but remember I don't know Anything about Electron, NPM, etc..
I'm a C/C++/Python Developer that uses Javascript only client-side!!

(But that wants to continue making beautufil presensations using Sozi.. )

Just installed Release v22.02.03. It has general Linux archive. and works fine.. So I'm okay for now just.. very worried..

Aussi: Je parle Français, mais attn mon orthographe/grammaire no sont pas au point car je ne l'écris quasimment jamais.

**Comment by aumouvantsillage** [comment]
Hi Frank.
In the latest version, I naively considered that the AppImage would be a good solution for distributions that do not support Debian, RPM or ArchLinux packages. It did not occur to me that tarballs would still be needed.
When you write:

> I don't have/want to infect my system with things like like Docker or AppImage

I don't understand what "infect" means in this context. The AppImage is a standalone executable and will not pollute your filesystem.

```
frank@minerva:~/src/ext/Sozi (master) $ npm install

npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: '@es-joy/jsdoccomment@0.10.8',
npm WARN EBADENGINE   required: { node: '^12 || ^14 || ^16' },
npm WARN EBADENGINE   current: { node: 'v18.6.0', npm: '8.13.2' }
npm WARN EBADENGINE }
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: 'eslint-plugin-jsdoc@36.1.1',
npm WARN EBADENGINE   required: { node: '^12 || ^14 || ^16' },
npm WARN EBADENGINE   current: { node: 'v18.6.0', npm: '8.13.2' }
npm WARN EBADENGINE }
```

According to these messages, your version of node.js is too recent.
Sozi depends on old versions of some modules. I will check whether they can be updated.

**Reply by BlueStilton** [reply]
Thanks, [aumouvantsillage](https://github.com/aumouvantsillage), I'll check out AppImage.  Maybe I'm too old and grumpy about these things. :_) By 'Infect' in this context I mean "render inconsistent with the package manager".

**Reply by BlueStilton** [reply]
And.. you are quite right. 

I didn't know AppImage and thought it was another custom virtualisation pre-packaging that required me to change my installs in may undesired ways (like Docker).  I downloaded the latest version as .appimage and it runs fine.

STILL: I don't like binary distribution (for auditability and out of principle). I'd be willing to create and maintain a Gentoo .ebuild if building from code would be more or less predictable, e.g. with the newer dependencies.

Gentoo is a rolling distro, and versions are often newer than expected.

All this is now nice-to-have for me. Since Sozi is the only application I'll maintain in this way, it's not a big deal to download an .appimage every now and then.

But if you'd like an ebuild at no maintance cost to you.. let me know :-)

Merci!
-f
